### PR TITLE
Port System.Security.Cryptography.X509Certificates to release/6.0

### PR DESCRIPTION
**Pull request by jonfortescue**
Port https://github.com/dotnet/arcade/pull/9064 to release/6.0.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by MattGal** [comment]
@ulisesh has been looking at this in parallel, you should synch as what's in main may not be sufficient yet.

**Review by MattGal — DISMISSED** [review]

File: src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj [hunk 1]
```diff
@@ -29,6 +29,7 @@
     <PackageReference Include="System.Net.Http" Version="$(SystemNetHttpVersion)" />
     <PackageReference Include="System.Collections.Immutable" Version="$(SystemCollectionsImmutableVersion)" />
     <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
+    <PackageReference Include="System.Security.Cryptography.X509Certificates" Version="$(SystemSecurityCryptographyX509CertificatesVersion)" />
```
Comment (ulisesh) [inline src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj:32]: You don't need to import System.Security.Cryptography.X509Certificates in every package, the change in Directory.Build.props does this already

Comment (jonfortescue) [reply src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj:32]: that's true, but this is just making identical changes to what's in main. If you want, I can drop this commit and keep just Matt's change.

Comment (ulisesh) [reply src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj:32]: Matt's change should be enough for CG

**Comment by ulisesh** [comment]
What's next for this PR @jonfortescue?

**Comment by jonfortescue** [comment]
@ulisesh I think this can be merged as is since I am going off FR and we know it resolves the issue it's supposed to.

**Comment by ulisesh** [comment]
@jonfortescue can you please hand it off to someone in the FR rotation?

**Comment by AlitzelMendez** [comment]
@ulisesh is something missing on this pr? or is ready to be approved/merged?

**Comment by ericstj** [comment]
This change should be undone.  It's causing issues downstream as it's introducing package dependencies in places where they aren't compatible.

See https://dev.azure.com/dnceng/public/_build/results?buildId=1770224&view=logs&j=108d2c4a-8a62-5a58-8dad-8e1042acc93c&t=f1671469-0c48-5b47-9674-85d49bdb5829&l=59

The fix for CG flagging this package is to address the reference higher in the stack, and only in the projects that cause the problem.

**Comment by MattGal** [comment]
> This change should be undone. It's causing issues downstream as it's introducing package dependencies in places where they aren't compatible.
> 
> See https://dev.azure.com/dnceng/public/_build/results?buildId=1770224&view=logs&j=108d2c4a-8a62-5a58-8dad-8e1042acc93c&t=f1671469-0c48-5b47-9674-85d49bdb5829&l=59
> 
> The fix for CG flagging this package is to address the reference higher in the stack, and only in the projects that cause the problem.

I'll get this taken care of today and make sure you're on the PR.

**Pull request merged**